src/utilities/alfalfa/AlfalfaPoint.cpp

```diff
@@ -0,0 +1,137 @@
+    return m_impl->m_input;
+  }
+
+  void AlfalfaPoint::setOutput(const AlfalfaComponent& component) {
+    if (component.canOutput()) {
+      m_impl->m_output = component;
+    } else {
+      throw std::runtime_error(fmt::format("Component of type: {} cannot be used as an output.", component.type));
+    }
+  }
+
+  boost::optional<AlfalfaComponent> AlfalfaPoint::output() const {
+    return m_impl->m_output;
+  }
+
+  void AlfalfaPoint::setUnits(const std::string& units) {
+    m_impl->m_units = units;
+  }
+
+  boost::optional<std::string> AlfalfaPoint::units() const {
+    return m_impl->m_units;
+  }
+
+  boost::optional<std::string> AlfalfaPoint::id() const {
+    boost::optional<std::string> result = m_impl->m_id;
+    if (!result.is_initialized()) {
+      std::string id = toIdString(displayName());
+      if (isValidId(id)) {
+        result = id;
+      } else {
+        LOG(Warn, fmt::format("Display name '{}' does not produce a valid point ID. Manually set a valid ID or export will fail.", displayName()));
+      }
+    }
+    return result;
+  }
+
+  void AlfalfaPoint::setId(const std::string& id) {
+    if (isValidId(id)) {
+      m_impl->m_id = id;
+    } else {
+      throw std::runtime_error(ID_VALID_CHARS_MSG);
+    }
+  }
+
+  std::string AlfalfaPoint::displayName() const {
+    return m_impl->m_display_name;
+  }
+
+  void AlfalfaPoint::setDisplayName(const std::string& display_name) {
+    if (display_name.size() == 0) {
+      throw std::runtime_error("Display name must have non-zero length");
+    }
+    m_impl->m_display_name = display_name;
+    if (!m_impl->m_id.is_initialized()) {
+      std::string id = toIdString(display_name);
+      if (!isValidId(id)) {
+        LOG(Warn, fmt::format("Display name '{}' does not produce a valid point ID. Manually set a valid ID or export will fail.", display_name));
+      }
+    }
+  }
+
+  bool AlfalfaPoint::optional() const {
+    return m_impl->m_optional;
+  }
+
+  void AlfalfaPoint::setOptional(bool optional) {
+    m_impl->m_optional = optional;
+  }
+
+  Json::Value AlfalfaPoint::toJSON() const {
+    Json::Value point;
+
+    if (id().is_initialized()) {
+      point["id"] = id().get();
+    } else {
+      throw std::runtime_error("Point requires a valid ID for export. " + ID_VALID_CHARS_MSG);
+    }
+    point["name"] = displayName();
+    if (input().is_initialized()) {
+      point["input"]["type"] = input().get().type;
+      point["input"]["parameters"] = input().get().parameters;
+    }
+
+    if (output().is_initialized()) {
+      point["output"]["type"] = output().get().type;
+      point["output"]["parameters"] = output().get().parameters;
+    }
+
+    if (units().is_initialized()) {
+      point["units"] = units().get();
+    }
+
+    point["optional"] = optional();
+
+    return point;
+  }
+
+  bool AlfalfaPoint::isValidId(const std::string& id) const {
+    return boost::regex_match(id, boost::regex("^[A-Za-z0-9_\\-\\[\\]:()]*$")) && id.size() > 0;
```

Comment:
Just an FYI but using a raw string litteral you can avoid the double slashing which improves readability a lot IMHO.

```suggestion
    return boost::regex_match(id, boost::regex(R"(^[A-Za-z0-9_\-\[\]:()]*$)")) && id.size() > 0;

```

Reply:
Also worth noting that boost::regex is truly awful in terms of performance (like std::regex), so if you expect this to be used a lot, make it a static regex.